python/ray/data/dataset.py

```diff
@@ -145,6 +146,7 @@
 TorchBatchType = Union[Dict[str, "torch.Tensor"], CollatedData]
 
 BT_API_GROUP = "Basic Transformations"
+J_API_GROUP = "Joining Operation APIs"
```

Comment:
should this be part of SMD_API_GROUP?

Reply:
Yeah, i can add to that group but i don't really understand the intention behind this grouping as it seems practically arbitrary in many cases

Reply:
I believe primarily it's that it gets rendered for documentation for better user discovery.